### DISCUSSION TITLE
cs_tools tools scriptability export - unexpected keyword argument: 'description'

### DISCUSSION THREAD

**Discussion by bmofkn** [thoughtspot/cs_tools#96] [Q&A] [answered]
We continue to use and love CS Tools on a daily basis!  We are running CS Tools v1.4.15 and have been since it's release.  Our cluster was upgraded to v9.8 last night.  

This morning, when executing `cs_tools tools scriptability export --include-types answer` and `cs_tools tools scriptability export --include-types liveboard` we received the following error for several answers:

`cs_tools.cli.tools.scriptability._export - ERROR - Parameter.__init__() got an unexpected keyword argument 'description'`

Did something change in ThoughtSpot v9.8.0.cl-213 that would suddenly require Parameters to have a description?  The Parameter UI indicates the description is optional.  Or perhaps CS Tools v1.4.15 was not anticipating this Parameter change from v9.8?

Thanks!

<img width="843" alt="CS Tools v1 4 15" src="https://github.com/thoughtspot/cs_tools/assets/119008346/cabdb67c-2975-4f47-8d48-644546c36261">

**Comment by boonhapus** [comment, marked as answer]
Hey @bmofkn , you're definitely correct. ThoughtSpot made **[a single change to the Parameter spec in TML](https://github.com/thoughtspot/thoughtspot_tml/compare/v2.0.11...v2.0.12)** for version 9.8. I've updated the underlying `thoughtspot_tml` library to match it.

We have two options.. either I can push a **CS Tools v1.4.16** that enforces this change, or if you're comfortable.. you can follow the below steps.

> 1. activate the virtual environment path from your above screenshot (`cs_tools self info`)
>     - source `"/Users/.../cs_tools/.cs_tools/bin/activate"`
> 2. type `pip install --upgrade thoughtspot_tml`
> 3. optionally, run `pip list | grep thoughtspot_tml` and verify that it installed v2.0.12
> 4. type `deactivate`
> ​

**Reply by bmofkn** [reply]
@boonhapus , your `pip install` did the trick!  All better now.  Thanks for the recommendation 👍
